### ISSUE TITLE
[apiv2] only outputs container ID whatever running any command inside a running container 

### ISSUE THREAD

**Issue by chuanchang** [containers/podman#6242]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

**Steps to reproduce the issue:**

1. ./bin/podman system service --timeout 0 tcp:127.0.0.1:8080

2. ./bin/podman run --name myalpine -td docker.io/library/alpine:latest /bin/sh

3. curl -H 'Content-type: application/json' -d '{"AttachStderr": true,"AttachStdin": true,"AttachStdout": true,"Cmd":["ps"]}' http://localhost:8080/v1.40/libpod/containers/my
alpine/exec

**Describe the results you received:**
```
[fedora@fedora32 libpod]$ curl -v -s -H 'Content-type: application/json' -d '{"AttachStderr": true,"AttachStdin": true,"AttachStdout": true,"Cmd":["ps"]}' http://localhost:8080/v1.40/libpod/contain
ers/myalpine/exec
*   Trying ::1:8080...
* connect to ::1 port 8080 failed: Connection refused
*   Trying 127.0.0.1:8080...
* Connected to localhost (127.0.0.1) port 8080 (#0)
> POST /v1.40/libpod/containers/myalpine/exec HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.69.1
> Accept: */*
> Content-type: application/json
> Content-Length: 76
>
* upload completely sent off: 76 out of 76 bytes
* Mark bundle as not supporting multiuse
< HTTP/1.1 201 Created
< Content-Type: application/json
< Date: Fri, 15 May 2020 10:08:22 GMT
< Content-Length: 74
<
{"Id":"dab67b13ed811abffdc46f2c9440627c58752996bb5213db55d1c9a14c2e48ca"}
* Connection #0 to host localhost left intact

[fedora@fedora32 libpod]$ curl -v -s -H 'Content-type: application/json' -d '{"AttachStderr": true,"AttachStdin": true,"AttachStdout": true,"Cmd":["ls"]}' http://localhost:8080/v1.40/libpod/contain
ers/myalpine/exec
*   Trying ::1:8080...
* connect to ::1 port 8080 failed: Connection refused
*   Trying 127.0.0.1:8080...
* Connected to localhost (127.0.0.1) port 8080 (#0)
> POST /v1.40/libpod/containers/myalpine/exec HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.69.1
> Accept: */*
> Content-type: application/json
> Content-Length: 76
>
* upload completely sent off: 76 out of 76 bytes
* Mark bundle as not supporting multiuse
< HTTP/1.1 201 Created
< Content-Type: application/json
< Date: Fri, 15 May 2020 10:12:10 GMT
< Content-Length: 74
<
{"Id":"4161676c34490cb01a9856726fc406a6604c38665e6f3781694019a653d3af7a"}
* Connection #0 to host localhost left intact
```

**Describe the results you expected:**
can output correct command result 

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:            2.0.0-dev
RemoteAPI Version:  1
Go Version:         go1.14.2
Git Commit:         4611ff5ce4ff67a142363ab66e10944bfc40f860
Built:              Fri May 15 03:26:27 2020
OS/Arch:            linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.15.0-dev
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.15-1.fc32.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.15, commit: 33da5ef83bf2abc7965fc37980a49d02fdb71826'
  cpus: 4
  distribution:
    distribution: fedora
    version: "32"
  eventLogger: file
  hostname: fedora32
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.6.10-300.fc32.x86_64
  linkmode: dynamic
  memFree: 1144713216
  memTotal: 4118323200
  ociRuntime:
    name: crun
    package: crun-0.13-2.fc32.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.13
      commit: e79e4de4ac16da0ce48777afb72c6241de870525
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.0.0-1.fc32.x86_64
    version: |-
      slirp4netns version 1.0.0
      commit: a3be729152a33e692cd28b52f664defbf2e7810a
      libslirp: 4.2.0
  swapFree: 0
  swapTotal: 0
  uptime: 97h 59m 13.98s (Approximately 4.04 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /home/fedora/.config/containers/storage.conf
  containerStore:
    number: 2
    paused: 0
    running: 2
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.0.0-1.fc32.x86_64
      Version: |-
        fusermount3 version: 3.9.1
        fuse-overlayfs: version 1.0.0
        FUSE library version 3.9.1
        using FUSE kernel interface version 7.31
  graphRoot: /home/fedora/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 2
  runRoot: /run/user/1000/containers
  volumePath: /home/fedora/.local/share/containers/storage/volumes
version:
  Built: 1589513187
  GitCommit: 4611ff5ce4ff67a142363ab66e10944bfc40f860
  GoVersion: go1.14.2
  OsArch: linux/amd64
  RemoteAPIVersion: 1
  Version: 2.0.0-dev
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
[fedora@fedora32 libpod]$ git rev-parse HEAD
4611ff5ce4ff67a142363ab66e10944bfc40f860
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Openstack VM

**Comment by mheon** [comment]
Exec is only partially implemented right now; #5831 has most of the rest.

**Comment by mheon** [comment]
This should be working since #5831 is landed, so I'm going to close.

Resize is still pending on another PR from @jwhonce though.

**Comment by mheon** [comment]
Feel free to reopen if exec issues persist.

**Comment by chuanchang** [comment]
> 
> 
> This should be working since #5831 is landed, so I'm going to close.
> 
> Resize is still pending on another PR from @jwhonce though.

Got the same result like before and w/o command output. 

```shell
[fedora@fedora32 libpod]$ git rev-parse HEAD
1a93857acc4ee1e5a9213e2c22f12802d84cd277

$ curl -v -s -H 'Content-type: application/json' -d '{"AttachStderr": true,"AttachStdin": true,"AttachStdout": true,"Cmd":["ls"]}' http://localhost:8080/v1.40/libpod/containers/myctr/exec
*   Trying ::1:8080...
* connect to ::1 port 8080 failed: Connection refused
*   Trying 127.0.0.1:8080...
* Connected to localhost (127.0.0.1) port 8080 (#0)
> POST /v1.40/libpod/containers/myctr/exec HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.69.1
> Accept: */*
> Content-type: application/json
> Content-Length: 76
>
* upload completely sent off: 76 out of 76 bytes
* Mark bundle as not supporting multiuse
< HTTP/1.1 201 Created
< Content-Type: application/json
< Date: Tue, 07 Jul 2020 09:00:21 GMT
< Content-Length: 74
<
{"Id":"045a7e4a2b7457f6dd242c996db67b26efb65182189f6079f3129679a0200cac"}
* Connection #0 to host localhost left intact
```

**Comment by vrothberg** [comment]
Same when using the remote client:

> $ /bin/podman-remote --url tcp://127.0.0.1:8080 exec d8 ls

... no output.

**Comment by zhangguanzhang** [comment]
@chuanchang , you must create an exec session first
```
$ id=`curl -s \
  -H 'Content-type: application/json' \
  -d '{"AttachStderr": true,"AttachStdin": true,"AttachStdout": true,"Cmd":["ls"]}' \
  http://localhost:8081/v1.40/libpod/containers/myctr/exec | jq -r .Id `
```
and start it
```
$ curl -o - -X POST http://localhost:8081/v1.40/libpod/exec/${id}/start -d '{"Detach":false,"Tty":false}' -H 'Content-type: application/json'
sbin
dev
docker-entrypoint.d
docker-entrypoint.sh
etc
home
lib
media
mnt
opt
proc
root
run
sbin
srv
sys
tmp
usr
var
[conmon:d]: exec with attach is waiting for start message from parent
[conmon:d]: exec with attach got start message from parent
```
and the version info
```
$ podman --version
podman version 2.0.4
```
